kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java

```diff
@@ -94,6 +107,17 @@ public Predicate(String name, List<Expression> children) {
         super(name, children);
     }
 
+    /** Constructor for a unary Predicate expression */
+    public Predicate(String name, Expression child) {
```

Comment:
Wondering if we need these two new constructors? It adds two more public APIs.

Reply:
We could add utility methods to `internal.ExpressionUtils` to make the Predicate create simple.

Reply:
I can move it to the `ExpressionUtils` if you'd like (although I think we'd be needing these in `kernel-api` for the data skipping) but I also think these are okay public APIs to expose? They make sense and it's likely connectors would want to use these constructors anyways.